### DISCUSSION TITLE
ST75256 pinout to Arduino ?

### DISCUSSION THREAD

**Discussion by Gudzys** [olikraus/u8g2#2032] [General]
Which pin is Clock or RW, Data, mosi. Reset is 9. Chip Select is 8 or 10?

Thanks.
![Screenshot_20221130-204057_DuckDuckGo](https://user-images.githubusercontent.com/119560681/204976064-ee2ae44a-bd8c-4620-935e-b2deff544a13.jpg)

**Comment by olikraus** [comment]
Do you mean st75256? How is your PCB configured related to the interface?

**Reply by Gudzys** [reply]
Thanks. Don't have the ST75256 yet; just trying to see which 4 pins of the 20 would be used re: clock, reset, RW/data/mosi, & chip select. Perhaps I should wait and see what connector the LCD arrives with, but from reading your notes there are only four of the 20 pins (besides power & ground) that need to be connected to Arduino: D13 (clock), D11 (RW/data/mosi), D10 (RS/chip select) & D8 (reset). Thanks again.

**Reply by olikraus** [reply]
Arduino side is simple: You can use any pins (although I mostly use the above mentioned pins for Arduino). However the display controller expects certain signals at specific pins. I once created an overview:
https://github.com/olikraus/u8g2/blob/master/doc/controller_interface.txt
You controller is mentioned in line 25. This information is taken from the datasheet of the ST75256 and yes, you need to connect all three pins D1, D2 and D3 with the data line (MOSI). 
Actually you need to connect five lines:

| Description | Suggested Arduino Pin | Display Pin |
|---|---|---|
| SPI Clock | D13 | D0 |
| MOSI | D11 | D1, D2, D3 |
| Chip Select | D10 | CS |
| Register Select, Command/Data | D9 | A0 (RS) |
| Display Reset | D8 | RES |

Note, that there are three names for the Register Select line:
RS = A0 = DC

In the datasheets you will find any of the three names, but they all mean the same, see also here: https://github.com/olikraus/u8g2/wiki/u8g2setupcpp#wiring

This means: Do not get confused between CS and RS. They denote two different signals and both are required.

When ordering your display, you usually need to order the display pre-configured for a specific interface (SPI, I2C, parallel, etc).
If you have ordered a display, configured for parallel interface, but you want SPI, then some solder bridges need to be modified. In most cases there are hints regarding this on the PCB (look here for an example: https://github.com/olikraus/u8g2/wiki/gallery#20-jun-2021-uc1638-lg196962)

**Reply by Gudzys** [reply]
How the Elegoo LCD came pre-pinned ready to connect to the breadboard. Would this work for the ST75256 & ST7565? Thanks.

![20221204_080204](https://user-images.githubusercontent.com/119560681/205504135-ca0fda37-05bb-43c9-b440-424dd6bcb1de.jpg)

**Reply by olikraus** [reply]
Sure, if you apply the correct wiring.

**Comment by Gudzys** [comment]
Thank you tremendously. Will see how it goes when I get the device.

Will also need a larger board to run the u8g2 programs.

Thanks again, so very much appreciated.

Brent

Sent with [Proton Mail](https://proton.me/) secure email.

------- Original Message -------

**Comment by Gudzys** [comment]
When you send 3 different data lines from the LCD to 1 Arduino input does the Arduino MCU parse each data cache from each of the 3 data lines or is there a loss of complexity on the Arduino side of things?

Thanks.

Sent with [Proton Mail](https://proton.me/) secure email.

------- Original Message -------

**Reply by olikraus** [reply]
You need to connect three pins from the display to one pin of the Arduino

**Reply by Gudzys** [reply]
Am using the breadboard to connect LCD ST75256 D1/D2/D3 to the same line and then connect then end of the line on the breadboard to Arduino D11; is this sufficient to combine the data from 3 pins into 1 Arduino pin? So far it is not working for me; perhaps there is some other problem that I can't see. 

Thanks.

**Reply by olikraus** [reply]
Maybe. Did you connect the other pins auf the display as specified in the datasheet?

**Comment by Gudzys** [comment]
What does this mean? Thanks.

--- do not connect
LHZ 3.3V, GND or unconnected
L/H connect to 3.3V or GND
3V3 connect to 3.3V
GND connect to ground (0V)
CLK SPI ClockDAT SPI Data

Sent with [Proton Mail](https://proton.me/) secure email.

------- Original Message -------

**Reply by olikraus** [reply]
I just quoted the information from the datasheet. Note: data is sent from Arduino to display

**Comment by Gudzys** [comment]
Yes I did, checking it twice & thrice. You mentioned a solder bridge for a parallel connection: have no idea if this is SPI or parallel. Thanks.

Sent with [Proton Mail](https://proton.me/) secure email.

------- Original Message -------

**Reply by olikraus** [reply]
> have no idea if this is SPI or parallel.

Well, then you need to figure out: There is no common suggestion or solution. The datasheet of your display should contain all the details. I could have a look on it, but you also didn't send a picture of your PCB.

**Comment by Gudzys** [comment]
It's ok. Thanks. Got it to flicker when the data arrives.  It's the 20-pin data at the top of this discussion.

**Reply by olikraus** [reply]
Sure, but the solder bridge are on the other side

**Comment by Gudzys** [comment]
[20221222_145808.jpg]
Thanks.

Sent with [Proton Mail](https://proton.me/) secure email.

------- Original Message -------

**Reply by olikraus** [reply]
I don't see an image here if you wanted to send one.

**Comment by Gudzys** [comment]
![20221222_145808](https://user-images.githubusercontent.com/119560681/209368750-37c2fc30-7c66-4a8c-869d-f8be55ee341a.jpg)

Sorry. Thanks.

**Reply by olikraus** [reply]
Nice. As you can see, there are several solder Bridges. R3, R5 and R7 are closed with a 0 ohm resistor. This is S configuration, which probably means 4 wire SPI.

**Comment by olikraus** [comment]
You could also post a picture of your complete wiring, so that we can crosscheck it here.

**Reply by Gudzys** [reply]
Pin 18 (D0) to Arduino D13
Pin 17/16/15 to Arduino D11 (these have been soldered together)
Pin 10 to Arduino D10
Pin 9 to Arduino D8
Pin 8 to Arduino D9
Pin 6 to Ground
Pin 5 to 5v

**Reply by olikraus** [reply]
Pin 5 is not required but you need 3.3V at pin 7.
What about the other data pins? You should follow the datasheet and connect them to 3.3V
I also suggest to connect /RD and /WR to 3.3V
Additionally you should add a 1uF cap between VSS and VDD
You should make a real complete list...

**Comment by Gudzys** [comment]
Won't light-up with pin 7, only with pin 5. 

Everything is connected except the pins 1-4. Will look for a 1uF cap.

Thanks.

**Comment by olikraus** [comment]
> Won't light-up with pin 7, only with pin 5.

pin 7 will light up the background LED (which is somehow irrelevant). 
pin 5 will power up the graphics controller ST75256 of the display. Without the graphics controller you can only see some boring backlight. Use 3.3v for pin  5!

**Comment by Gudzys** [comment]
Thank you & thanks again for all your help with this.

Sent with [Proton Mail](https://proton.me/) secure email.

------- Original Message -------
